catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/SourceOperations.java

```diff
@@ -296,82 +275,51 @@ boolean isSourceAvailable(Source source) {
-      Set<String> ids = sourceInfoRequest.getSourceIds();
-
-      // Only return source descriptor information if this sourceId is
-      // specified
-      if (ids != null) {
-        Optional<String> notLocal = ids.stream().filter(s -> !s.equals(getId())).findFirst();
-        if (notLocal.isPresent()) {
-          SourceUnavailableException sourceUnavailableException =
-              new SourceUnavailableException("Unknown source: " + notLocal.get());
-          LOGGER.debug(
-              "Throwing SourceUnavailableException for unknown source: {}",
-              notLocal.get(),
-              sourceUnavailableException);
-          throw sourceUnavailableException;
-        }
-      }
-
-      // Fanout will only add one source descriptor with all the contents
-      Set<Source> availableSources =
-          frameworkProperties
-              .getFederatedSources()
-              .values()
-              .stream()
-              .map(source -> frameworkProperties.getSourcePoller().getCachedSource(source))
-              .filter(source -> source != null && source.isAvailable())
-              .collect(Collectors.toSet());
-
-      Set<ContentType> contentTypes =
-          availableSources
-              .stream()
-              .map(source -> frameworkProperties.getSourcePoller().getCachedSource(source))
-              .filter(source -> source.getContentTypes() != null)
-              .map(Source::getContentTypes)
-              .flatMap(Collection::stream)
-              .collect(Collectors.toSet());
-
-      Source localSource = null;
-
-      if (catalog != null) {
-        localSource = frameworkProperties.getSourcePoller().getCachedSource(catalog);
-        if (localSource != null && localSource.isAvailable()) {
-          availableSources.add(localSource);
-          contentTypes.addAll(localSource.getContentTypes());
-        }
+    // Only return source descriptor information if this sourceId is
+    // specified
+    if (ids != null) {
+      Optional<String> notLocal = ids.stream().filter(s -> !s.equals(getId())).findFirst();
+      if (notLocal.isPresent()) {
+        SourceUnavailableException sourceUnavailableException =
+            new SourceUnavailableException("Unknown source: " + notLocal.get());
+        LOGGER.debug(
+            "Throwing SourceUnavailableException for unknown source: {}",
+            notLocal.get(),
+            sourceUnavailableException);
+        throw sourceUnavailableException;
       }
+    }
 
-      List<Action> actions = getSourceActions(localSource);
-
-      // only reveal this sourceDescriptor, not the federated sources
-      sourceDescriptor = new SourceDescriptorImpl(this.getId(), contentTypes, actions);
-      if (this.getVersion() != null) {
-        sourceDescriptor.setVersion(this.getVersion());
+    // Fanout will only add one source descriptor with all the contents
+    final Set<Source> sources = new HashSet<>(frameworkProperties.getFederatedSources().values());
```

Comment:
Do we need to add the local catalog to this list of sources? Looks like it was added previously.

Reply:
I think that's done in the line below. The availability and content types are handled later with all the other sources so the functionality should be the same.